### PR TITLE
fixes #7352 - missing DNS record returns 404

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

File: modules/dns/providers/virsh.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def find_ip_for_host host
           return e.parent.attributes["ip"]
         end
       end
-      raise Proxy::Dns::Error.new("Cannot find DNS entry for #{host}")
+      raise Proxy::Dns::NotFound.new("Cannot find DNS entry for #{host}")
```
Comment (dmitri-d) [inline modules/dns/providers/virsh.rb:58]: Not going to work here, as the exception will be recast as `Proxy::Dns::Error`.

**Comment by dmitri-d** [comment]
@shlomizadok: ping?

**Comment by dmitri-d** [comment]
looks good; @shlomizadok, could you add tests that verify we throw `Proxy::Dns::NotFound` exception when record isn't found?

**Comment by shlomizadok** [comment]
@witlessbird , added some tests for a beginning. and all are :green_apple:

**Comment by dmitri-d** [comment]
merged at 3f4ee4b. Thanks, @shlomizadok!

**Pull request closed without merging**